ncutils/netclientutils.go

```diff
@@ -158,6 +158,37 @@ func GetPublicIP(api string) (net.IP, error) {
 	return net.ParseIP(endpoint), err
 }
 
+// GetPublicIPv6 - gets public ipv6 address
+func GetPublicIPv6() (net.IP, error) {
```

Comment:
can't we use stun to get a public endpoint?
also if a machine has dual-stack network, what's the guarantee that this code will fetch the ipv6 endpoint not ipv4?

Reply:
@abhishek9686 , as tested, currently stun func returns IPv4 address for  a dual-stack network host.  For getting the IPv6 address, it will need to make changes as well.
The new added GetPublicIPv6 func, it returns IPv6 address for a dual-stack network host. It's already tested.